www/js/controlHelper.ts

```diff
@@ -0,0 +1,136 @@
+ * @param fileName is the name of the file to be created
+ * @returns a function that returns a promise, which writes the file upon evaluation.
+ */
+export const createWriteFile = function (fileName: string) {
+  return function(result: RawDataCluster) {
+    const resultList = result.phone_data;
+      return new Promise<void>(function(resolve, reject) {
+        window.requestFileSystem(window.LocalFileSystem.TEMPORARY, 0, function(fs) {
+          logDebug(`file system open: ${fs.name}`);
+          fs.root.getFile(fileName, { create: true, exclusive: false }, function (fileEntry) {
+            logDebug(`fileEntry ${fileEntry.nativeURL} is file? ${fileEntry.isFile.toString()}`)
+            fileEntry.createWriter(function (fileWriter) {
+              fileWriter.onwriteend = function() {
+                logDebug("Successful file write...");
+                resolve();
+              }
+              fileWriter.onerror = function(e) {
+                logDebug(`Failed file write: ${e.toString()}`);
+                reject();
+              }
+
+              // if data object is not passed in, create a new blog instead.
+              const dataObj = new Blob([JSON.stringify(resultList, null, 2)],
+                { type: "application/json" });
+              fileWriter.write(dataObj);
+            })
+          });
+        });
+    });
+}};
+
+/**
+ * createShareData returns a shareData method, with the input parameters captured.
+ * @param fileName is the existing file to be sent
+ * @param startTimeString timestamp used to identify the file
+ * @param endTimeString " "
+ * @returns a function which returns a promise, which shares an existing file upon evaluation.
+ */
+export const createShareData = function(fileName: string, startTimeString: string, endTimeString: string) {
+  return function() {
+    return new Promise<void>(function(resolve, reject) {
+    window.requestFileSystem(window.LocalFileSystem.TEMPORARY, 0, function(fs) {
+      logDebug("During email, file system open: " + fs.name);
+      fs.root.getFile(fileName, null, function(fileEntry) {
+        logDebug(`fileEntry ${fileEntry.nativeURL} is file? ${fileEntry.isFile.toString()}`);
+        fileEntry.file(function(file) {
+          const reader = new FileReader();
+
+          reader.onloadend = function() {
+            const readResult = this.result as string;
+            logDebug(`Successfull file read with ${readResult.length} characters`);
+            const dataArray = JSON.parse(readResult);
+            logDebug(`Successfully read resultList of size ${dataArray.length}`);
+            let attachFile = fileEntry.nativeURL;
+              if (window['device'].platform === "android")
+                attachFile = "app://cache/" + fileName;
+              if (window['device'].platform === "ios")
+                alert(i18next.t("email-service.email-account-mail-app"));
+              const email = {
+                'files': [attachFile],
```

Comment:
I'm surprised this works with socialsharing. IIRC, I had to create a PR to have the email plugin support the `app://` prefix. Have you tried actually emailing/saving the related file? Does it generate a non-blank file?

Reply:
Yes! As written, this works fine - but after a quick test, it also works without the `app://` prefix, so I'll go ahead and remove it.